src/reducers/existingThoughtMove.js

```diff
@@ -180,11 +180,24 @@ export default (state, { oldPath, newPath, offset }) => {
     }
   })
 
+  /**
+   * Keep cursor position on child element after moving parent
+   */
+  const generateCursorNewPath = () => {
+    if (editing) {
+      const children = (state.cursor || []).slice(oldPath.length)
```

Comment:
Let's call this `cursorDescendantPath`. Children refer to the subthoughts beneath a parent thought. Here rather you are getting the slice of the path that descends from the cursor.